src/components/PurchaseOrder/PO.js

```diff
@@ -155,21 +158,44 @@ class PO extends Component {
     </Button>
   );
 
+  mountModal = () => {
+    this.setState({ isRenderModal: true });
+  }
+
+  unmountModal = () => {
+    this.setState({ isRenderModal: false });
+  }
+
   render() {
     const { location, history, match, mutator, resources, parentResources } = this.props;
     const order = get(resources, ['order', 'records', 0]);
     const orderId = get(order, 'id');
     const orderNumber = get(order, 'po_number', '');
     const poLines = get(order, 'po_lines', []);
+    const workflowStatus = get(order, 'workflow_status', '');
+    const isVisible = orderId && workflowStatus === WORKFLOW_STATUS.open;
     const lastMenu = (
       <PaneMenu>
         <IfPermission perm="purchase_order.item.put">
-          <CloseOrderModal
-            closeOrderSubmit={this.closeOrder}
-            closingReasons={resources.closingReasons.records}
-            orderId={orderId}
-            workflowStatus={get(order, 'workflow_status')}
-          />
+          {isVisible && (
+            <Button
+              buttonStyle="primary"
+              marginBottom0
+              onClick={() => this.mountModal()}
+              style={{ marginRight: '10px' }}
+            >
+              <FormattedMessage id="ui-orders.paneBlock.closeBtn" />
+            </Button>
+          )}
+          {this.state.isRenderModal && (
+            <CloseOrderModal
+              closeOrderSubmit={this.closeOrder}
+              closingReasons={resources.closingReasons.records}
+              orderNumber={orderNumber}
+              isOpen={this.state.isRenderModal}
```

Comment:
does this prop make sense anymore? As I can see, rendering of modal is controlled by state property `isRenderModal`

Reply:
updated